src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -370,45 +457,124 @@ def set_ior_location(self, param_type, path, pool=None, cont=None,
+            self.mdtest_cmd.api.update("DFS", display_api)
+            self.mdtest_cmd.test_dir.update(path, display_test_dir)
+            if pool and cont_uuid:
+                self.mdtest_cmd.set_daos_params(self.server_group,
+                                                pool, cont_uuid)
+            elif pool:
+                self.mdtest_cmd.set_daos_params(self.server_group,
+                                                pool, None)
+
+    def run_mdtest_with_params(self, param_type, path, pool=None, cont=None,
+                               flags=None, display=True):
+        """Set the mdtest params and run mdtest.
+
+        Args:
+            param_type: see set_ior_params
+            path: see set_mdtest_params
+            pool: see set_mdtest_params
+            cont: see set_mdtest_params
+            flags see set_mdtest_params
+            display (bool, optional): print updated params. Defaults to True.
+        """
+        self.set_mdtest_params(param_type, path, pool, cont, flags, display)
+        self.run_mdtest(self.get_mdtest_job_manager_command(self.manager),
+                        self.mdtest_processes,
+                        display_space=(True if pool else False), pool=pool)
+
     def run_datamover(self, test_desc=None,
+                      src_type=None, src_path=None,
+                      src_pool=None, src_cont=None,
+                      dst_type=None, dst_path=None,
+                      dst_pool=None, dst_cont=None,
                       expected_rc=0, expected_output=None,
                       processes=None):
-        """Run the DataMover command.
-
-        Currently, this only uses "dcp".
+        """Run the corresponding command specified by self.tool.
+        Calls set_datamover_params if and only if any are passed in.
 
         Args:
             test_desc (str, optional): description to print before running
+            src_type: see set_datamover_params
+            src_path: see set_datamover_params
+            src_pool: see set_datamover_params
+            src_cont: see set_datamover_params
+            dst_type: see set_datamover_params
+            dst_path: see set_datamover_params
+            dst_cont: see set_datamover_params
             expected_rc (int, optional): rc expected to be returned
             expected_output (list, optional): substrings expected to be output
             processes (int, optional): number of mpi processes.
-                defaults to self.processes
+                defaults to self.dcp_processes
 
         Returns:
             The result "run" object
 
         """
-        if not processes:
-            processes = self.processes
+        self.num_run_datamover += 1
+        self.log.info("run_datamover called %s times",
+            str(self.num_run_datamover))
+
+        # Set the params if and only if any were passed in
+        if (src_type or src_path or src_pool or src_cont or
```

Comment:
(pylint-too-many-boolean-expressions) Too many boolean expressions in if statement (8/5)